Source/Methods/Envelope Tracking/EnvelopeTracking.cs

```diff
@@ -0,0 +1,309 @@
+        }
+
+        public struct SynchronizationConfiguration
+        {
+            public double RFDelayRange_s;
+            public double RFDelay_s;
+
+            public static SynchronizationConfiguration GetDefault()
+            {
+                return new SynchronizationConfiguration()
+                {
+                    RFDelayRange_s = 2e-6, // +/- 1us
+                    RFDelay_s = 0.0
+                };
+            }
+        }
+        #endregion
+
+        #region Envelope Creation
+        public static Waveform CreateEnvelopeWaveform(Waveform referenceWaveform, DetroughConfiguration detroughConfig)
+        {
+            Waveform envelopeWaveform = new Waveform()
+            {
+                Name = referenceWaveform.Name + "Envelope",
+                Data = referenceWaveform.Data.Clone(),
+                SignalBandwidth_Hz = 0.8 * referenceWaveform.SampleRate,
+                PAPR_dB = double.NaN, // unnecessary to calculate as it won't be used
+                BurstLength_s = double.NaN, // not applicable in ET
+                SampleRate = referenceWaveform.SampleRate,
+                // burst start and stop locations will be left as null since they also will not be used
+                IdleDurationPresent = referenceWaveform.IdleDurationPresent, // can't null a bool, so copy from reference waveform
+                RuntimeScaling = 10 * Math.Log10(0.9), // applies 10% headroom to the waveform at runtime
+                Script = referenceWaveform.Script // we will copy the script but call replace on it in the following lines to change the waveform name
+            };
+            envelopeWaveform.Script = envelopeWaveform.Script.Replace(referenceWaveform.Name, envelopeWaveform.Name); // this is better here since now we only have to add the suffix once
+            WritableBuffer<ComplexSingle> envWfmWriteBuffer = envelopeWaveform.Data.GetWritableBuffer();
+
+            double[] iqMagnitudes = referenceWaveform.Data.GetMagnitudeDataArray(false);
+            double detroughRatio = detroughConfig.MinimumVoltage / detroughConfig.MaximumVoltage;
+
+            // Waveforms are assumed to be normalized in range [0, 1], so no normalization will happen here
+            switch (detroughConfig.Type)
+            {
+                case DetroughType.Exponential:
+                    // Formula: e = IQmag + d*exp(-IQmag/d)
+                    double expScale = 1 + detroughRatio * Math.Exp(-1.0 / detroughRatio);
+                    for (int i = 0; i < envWfmWriteBuffer.Size; i++)
+                    {
+                        double sampleValue = iqMagnitudes[i] + detroughRatio * Math.Exp(-iqMagnitudes[i] / detroughRatio);
+                        envWfmWriteBuffer[i] = ComplexSingle.FromSingle((float)(detroughConfig.MaximumVoltage * sampleValue / expScale)); // Scale detrough to Vmax. Divide by detroughed waveform's max value to normalize. IQMagnitude's max value is 1

```

Comment:
Comment for lines 124, 133, and 142
```suggestion
                        // Scale detrough to Vmax. Divide by detroughed waveform's max value to normalize. IQMagnitude's max value is 1
                        envWfmWriteBuffer[i] = ComplexSingle.FromSingle((float)(detroughConfig.MaximumVoltage * sampleValue / expScale)); 
```
Let's move the comments onto the line above so we don't have such long lines. Same for 133 and 142.

Reply:
71e2582950478218751e3253aa3474fc62ebf279